CI fails when commit hashes starting with 0 are merged

So we just merged 0571889 and the build says:

```
> Run hack/verify-installation.sh
Extracting krew from tarball.
Installing the krew build to a temporary directory.
[...]

WARNING: Detected stdin, but discarding it because of --manifest or args
F0329 06:20:14.510198    9214 root.go:79] failed to load plugin manifest from 
file: plugin manifest validation error: failed to parse plugin version: illegal 
zero-prefixed version component "0571889" in "v0.0.0-detached.0571889"
##[error]Process completed with exit code 255.
```

It seems like 

```
v0.0.0-detached.0571889
                ^^^^^^^
````

this is somehow a problem –which I never expected.

The error is coming from `semver.Parse` –and I'm not sure if https://semver.org dictates that stuff that comes after the `-` are supposed to be validated like this. If so, maybe we need to switch to `+` syntax.

It would be great if someone can take a look

/help
/good-first-issue
/kind flake
/priority backlog

--- a/hack/make-release-artifacts.sh
+++ b/hack/make-release-artifacts.sh
@@ -71,7 +71,7 @@ git_describe="$(git describe --tags --dirty --always)"
 if [[ ! "${git_describe}" =~ v.* ]]; then
   # if tag cannot be inferred (e.g. CI/CD), still provide a valid
   # version field for krew.yaml
-  git_describe="v0.0.0-detached.${git_describe}"
+  git_describe="v0.0.0-detached+${git_describe}"
 fi
 krew_version="${TAG_NAME:-$git_describe}"
 cp ./hack/krew.yaml ./out/krew.yaml

